mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -210,13 +212,35 @@ pub trait MergeUtxoArcOps<T: UtxoCommonOps + GetUtxoListOps>: UtxoCoinBuilderCom
     }
 }
 
+macro_rules! sync_status_notice_on_error {
+    ($sync_status_loop_handle:ident, $delay: expr, $error: ident) => {
+        $sync_status_loop_handle.notify_on_temp_error($error.to_string());
+        Timer::sleep($delay).await;
+        continue;
+    };
+}
+
+macro_rules! block_header_utxo_loop_validate_headers {
+    ($coin: expr, $ticker: ident, $temporary_from: ident, $block_headers: expr, $storage: ident, $sync_status_loop_handle:ident) => {
+         if let Some(params) = &$coin.as_ref().conf.block_headers_verification_params {
+            if let Err(e) = validate_headers($ticker, $temporary_from, $block_headers, $storage, &params).await {
+                error!("Error {} on validating the latest headers!", e);
+                // Todo: remove this electrum server and use another in this case since the headers from this server are invalid
+                $sync_status_loop_handle.notify_on_permanent_error(e.to_string());
+                break;
+            }
+        }
+    };
+}
+
```

Comment:
I don't believe macros should be declared if they are used inside one function only, we tend to define a function inside the function in this case to avoid code reuse. If these cases don't apply (a function can't be defined for them) you can revert to the code duplication as before since there is better ways to refactor the code while avoiding duplication I believe. In my opinion we should call `retrieve_headers` and `validate_headers` only once in every loop, the passed values should differ depending on if we are gonna use chunks or not.

Reply:
I used macro to avoid DRY but now I understand this can be difficult to modify or even understand in the future so I'll just revert. Thanks

helper function can't work here btw